reactNative/android/src/main/cpp/SampleSource.cpp

```diff
@@ -14,43 +14,175 @@
+                    }
+                    outBuff[frameIndex] += buffer[frameIndex++] * mGain;
+                }
+            } else if ((sampleChannels == 1) && (numChannels == 2)) {
+                // STEREO output from MONO samples
+                int dstSampleIndex = 0;
+                for (int32_t frameIndex = 0; frameIndex < numWriteFrames * sampleChannels;) {
+                    mCurSampleIndex++;
+                    if (abs(buffer[frameIndex]) > amplitudeMax) {
+                        amplitudeMax = abs(buffer[frameIndex]);
+                    }
+                    outBuff[dstSampleIndex++] += buffer[frameIndex] * mLeftGain;
+                    outBuff[dstSampleIndex++] += buffer[frameIndex++] * mRightGain;
+                }
+            } else if ((sampleChannels == 2) && (numChannels == 1)) {
+                // MONO output from STEREO samples
+                int dstSampleIndex = 0;
+                for (int32_t frameIndex = 0; frameIndex < numWriteFrames * sampleChannels;) {
+                    mCurSampleIndex += 2;
+                    if (abs(buffer[frameIndex]) > amplitudeMax) {
+                        amplitudeMax = abs(buffer[frameIndex]);
+                    }
+                    if (abs(buffer[frameIndex + 1]) > amplitudeMax) {
+                        amplitudeMax = abs(buffer[frameIndex + 1]);
+                    }
+                    outBuff[dstSampleIndex++] += buffer[frameIndex++] * mLeftGain +
+                                                 buffer[frameIndex++] * mRightGain;
+                }
+            } else if ((sampleChannels == 2) && (numChannels == 2)) {
+                // STEREO output from STEREO samples
+                int dstSampleIndex = 0;
+
+                for (int32_t frameIndex = 0; frameIndex < numWriteFrames * sampleChannels;) {
+                    mCurSampleIndex += 2;
+                    if (abs(buffer[frameIndex]) > amplitudeMax) {
+                        amplitudeMax = abs(buffer[frameIndex]);
+                    }
+                    if (abs(buffer[frameIndex + 1]) > amplitudeMax) {
+                        amplitudeMax = abs(buffer[frameIndex + 1]);
+                    }
+                    outBuff[dstSampleIndex++] += buffer[frameIndex++] * mLeftGain;
+                    outBuff[dstSampleIndex++] += buffer[frameIndex++] * mRightGain;
+                }
+            }
+
+            if (mCurSampleIndex >= numSamples) {
+                mIsPlaying = false;
+            }
+        }
+
+        float logPower = fmax((float)MIN_DB, log10(amplitudeMax) * (float)10);
+        __android_log_print(ANDROID_LOG_INFO, TAG, "log power %f", logPower);
+        __android_log_print(ANDROID_LOG_INFO, TAG, "min decibels %f", mMinDecibels);
+        __android_log_print(ANDROID_LOG_INFO, TAG, "max decibels %f", mMaxDecibels);
+        float scaledPower = fmin((float)1, (logPower - mMinDecibels) / (mMaxDecibels - mMinDecibels));
+        mLastAmplitude = scaledPower * mGain;
```

Comment:
If we disregard the volume, we gain more flexibility with the UI, allowing it to decide whether to consider the volume or not. Currently, iOS returns the calculated amplitude without scaling it according to the set volume. So I suggest not to do it here too.

Reply:
I'm actually thinking if that makes sense. Have not decided yet, on which side to take the gain into account. Please disregards this comment for now.